### PR TITLE
Rearch: Add missing sections and indexes

### PR DESCRIPTION
**Pull request by talagluck**
Also added the data sources section and filled out the index pages.

For the individual Data Source pages, I pulled these from what we already had.

### CODE REVIEW

File: integrations/index.md [hunk 1]
```diff
@@ -5,4 +5,43 @@ nav_order: 4
...
+If you click the Connect button in Cloud, you will see a window which shows a few
+possibilities for connection strings. Select Postgres to get a connection string
+that you can use to connect.
+
+### GlareDB local
+
+Using GlareDB locally? You can spin up GlareDB as a local Postgres server using
+
+```shell
+./glaredb server --data-dir <path to your GlareDB instance>
```
Comment (scsmithr) [inline integrations/index.md:30]:
```suggestion
./glaredb server --data-dir <path to storage directory>
```

`--data-dir` just says to glaredb to persist its data in that directory.

File: cloud/index.md [hunk 2]
```diff
@@ -4,3 +4,7 @@ layout: default
 nav_order: 3
 has_children: true
 ---
+
+deployments
+user provisions
+billing
```
Comment (greyscaled) [inline cloud/index.md:10]:
```suggestion

# GlareDB Cloud

```

File: cloud/user-provisions.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+title: User Provisions
+layout: default
+nav_order: 2
+parent: GlareDB Cloud
+---
+
+## User Provisions
```
Comment (greyscaled) [inline cloud/user-provisions.md:8]: thought (non-blocking): I wonder if `Provisioning Users` is clearer, or even `Managing Users`.

Do we intend to make this just users in an org, or user/password credentials for accessing cloud?

Unsure based on title alone.

File: data-sources/azure.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+GlareDB can use CSV, ndjson, or Parquet files in Azure Blob Storage as external
+data sources.
+
+<!-- prettier-ignore-start -->
+
+- TOC
+{:toc}
+<!-- prettier-ignore-end -->
+
```
Comment (greyscaled) [inline data-sources/azure.md:24]:
```suggestion
# Azure Blob Storage

GlareDB can use CSV, ndjson, or Parquet files in Azure Blob Storage as external
data sources.

```

File: scaling/index.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+## Scaling
+
+- hybrid execution
+- distributed execution
+- embedded (in applications, in notebooks)
+- hybrid exec and hybrid workloads
+- brief intro to cloud
+- some highlevel stuff about materialized views
```
Comment (greyscaled) [inline scaling/index.md:15]:
```suggestion
- some high level stuff about materialized views
```

**Review by greyscaled — APPROVED** [review]

**Pull request merged**